benchmark/engine.py

```diff
@@ -83,13 +85,20 @@ def __str__(self):
         raise NotImplementedError()
 
     @classmethod
-    def create(cls, engine_type, gcp_project_id, ibm_model_id, ibm_custom_id):
+    def create(cls, **kwargs):
+        engine_type = NLUEngines[kwargs['engine_type']]
+        gcp_project_id = kwargs['gcp_project_id']
+        ibm_model_id = kwargs['ibm_model_id']
+        ibm_custom_id = kwargs['ibm_custom_id']
+        apikey = kwargs['apikey']
+        stt_url = kwargs['stt_url']

```

Comment:
you should look for keys only when constructing the engine of the specific type. what happens if you are constructing Lex and none of these are in the dictionary?

Reply:
You're right. The middle 3 variables would have None values, but I would definitely get a KeyError with the last two.